.github/workflows/build-webapi.yaml

```diff
@@ -9,27 +9,45 @@ on:
     branches: [ dev, main ]
     paths:
       - 'src/dotnet/**'
+      - '.github/workflows/**'
 
 jobs:
-  build:
+  container-build-step:
     runs-on: ubuntu-latest
-    defaults:
-      run:
-        working-directory: ./src/dotnet/CarbonAware.WebApi
     steps:
+    - name: Checkout Repository
+      uses: actions/checkout@v2
+    
+    - name: Docker Target Build
+      run: docker build --target=build . -f CarbonAware.WebApi/Dockerfile -t ca-api-build
+      working-directory: ./src/dotnet
+
+    - name: Docker Unit Test
+      run: docker run -w //src/ ca-api-build dotnet test
+      working-directory: ./src/dotnet
 
-    - uses: actions/checkout@v2
-    - name: Setup .NET
-      uses: actions/setup-dotnet@v1
-      with:
-        dotnet-version: 6.0.x
+  container-publish-step:
+      runs-on: ubuntu-latest
+      steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v2
 
-    - name: Restore dependencies
-      run: dotnet restore
+      - name: Docker Target Final
+        run: docker build . -f CarbonAware.WebApi/Dockerfile -t ca-api
+        working-directory: ./src/dotnet
+        
+      - name: Docker Run Container
+        run: |
+          docker run -d --name publish-container -p 8080:80 ca-api
+          docker container ls
+          
+      - name: Docker Wget in container
```

Comment:
Is there a better more extensible way we can separate the publish / build step from the validation step?

Reply:
@dnastrain 

We could, in theory, build the image, tarball it, upload it as an artifact, and then download it in a different job, unball it, and run it so we can validate. Or we could set up a container registry, log in, and publish the image. 

Short of those, no. Ideally, we would have a container registry/dockerhub account ready to go, but we haven't set up any infra yet as we don't know what the infra will look like and that's an open task we haven't picked up yet.